openstudiocore/src/model/ScheduleFile.hpp

```diff
@@ -0,0 +1,143 @@
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ *  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER, THE UNITED STATES GOVERNMENT, OR ANY CONTRIBUTORS BE LIABLE FOR
+ *  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ *  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ *  AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **********************************************************************************************************************/
+
+#ifndef MODEL_SCHEDULEFILE_HPP
+#define MODEL_SCHEDULEFILE_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ScheduleInterval.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+// TODO: Check the following class names against object getters and setters.
+class ScheduleTypeLimits;
+class ExternalFile;
+
+namespace detail {
+
+  class ScheduleFile_Impl;
+
+} // detail
+
+/** ScheduleFile is a ScheduleInterval that wraps the OpenStudio IDD object 'OS:Schedule:File'. */
+class MODEL_API ScheduleFile : public ScheduleInterval {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  ScheduleFile(const ExternalFile& externalfile, int column=1, int rowsToSkip=0);
+
+  virtual ~ScheduleFile() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> minutesperItemValues();
+
+  /** @name Getters */
+  //@{
+
+  // TODO: Check return type. From object lists, some candidates are: ScheduleTypeLimits.
+  boost::optional<ScheduleTypeLimits> scheduleTypeLimits() const;
+
+  ExternalFile externalFile() const;
+
+  int columnNumber() const;
+
+  int rowstoSkipatTop() const;
+
+  boost::optional<int> numberofHoursofData() const;
+
+  bool isNumberofHoursofDataDefaulted() const;
+
+  bool interpolatetoTimestep() const;
+
+  bool isInterpolatetoTimestepDefaulted() const;
+
+  boost::optional<std::string> minutesperItem() const;
+
+  bool isMinutesperItemDefaulted() const;
+
+  //@}
+  /** @name Setters */
+  //@{
+
+  // TODO: Check argument type. From object lists, some candidates are: ScheduleTypeLimits.
+  bool setScheduleTypeLimits(const ScheduleTypeLimits& scheduleTypeLimits);
+
+  bool resetScheduleTypeLimits();
+
+  bool setColumnNumber(int columnNumber);
+
+  bool setRowstoSkipatTop(int rowstoSkipatTop);
+
+  bool setNumberofHoursofData(int numberofHours);
+
+  bool setInterpolatetoTimestep(bool interpolatetoTimestep);
+
+  void resetInterpolatetoTimestep();
+
+  bool setMinutesperItem(const std::string& minutesperItem);
```

Comment:
Add a method taking an int?

Reply:
@jasondegraw and @macumber should the `setMinutesperItem` method take an `int` argument rather than a string? I wrote a measure in 3.1.0 and it only takes a string, which differs from the field in EP. If so, I can open an issue.

Reply:
Yeah that seems weird, it should be easy to add an override that takes an `int`

Reply:
I think this may be an unintended outcome of using a choice in the OpenStudio IDD here rather than what E+ does. But I agree, at the very least just adding an int version makes sense.